specs/psycopg_analysis.md

```diff
@@ -0,0 +1,176 @@
+# Analysis of psycopg package
+
+## Overview
+
+The Python package `psycopg` is a "py.typed" package with inlined type annotations.
+
+Users of Pylance reported that they were not receiving completion suggestions for certain symbols imported from psycopg. Running "pyright --verifytypes" on the library revealed that 349 of 1472 public symbols exported from the library were lacking type annotations and were therefore treated by Pyright’s logic as an `Unknown` type.
```

Comment:
Many things are not specific to psycopg; e.g., while trying to run `pyright --verifytypes` on [click](https://github.com/pallets/click/) (tried quickly, because this raises a *lot* or "errors"), I found a lot of similar issues.

Which makes me wonder if it'd be worth to have a "primer" similar to https://github.com/hauntsaninja/mypy_primer for this verifytypes?

(Note: I'm not a user of pyright myself, mostly coming from psycopg perspective.)